### PR TITLE
[docs] Adding list of supported RDBMS

### PR DESCRIPTION
**Pull request by john-bodley**
### CATEGORY

Choose one

- [ ] Bug Fix
- [ ] Enhancement (new features, refinement)
- [ ] Refactor
- [ ] Add tests
- [ ] Build / Development Environment
- [x] Documentation

### SUMMARY

Whilst modifying the database engine specs I thought there was merit in explicitly listing all the RDBMS Superset currently supports. I've also updated the SQLAlchemy dialect logic to be consistent (naming and completeness) with the list of supported databases.

### TEST PLAN

CI.

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### REVIEWERS

to: @betodealmeida @mistercrunch

### CODE REVIEW

**Comment by villebro** [comment]
I like the idea of this list; however, there are already two similar lists, one on the main `README.md` and the other on `installation.rst`. Perhaps we should remove the table from README and just provide a link to the official list on index? The installation one I think is good to keep, as it has technical details that would not sit well on the main index.

**Comment by john-bodley** [comment]
Thanks. I wasn't aware of the list in the `README.md` file.

File: docs/index.rst [hunk 1]
```diff
@@ -93,6 +93,32 @@ Features
 - Integration with most SQL-speaking RDBMS through SQLAlchemy
 - Deep integration with Druid.io
 
+The following RDBMS are currently suppored:
```
Comment (mistercrunch) [inline docs/index.rst]: [optional]: "Other database engines with a proper DBAPI driver and SQLAlchemy dialect should be supported as well."  (with the right links to DBAPI spec and SQLAlchemy definition)

Comment (john-bodley) [reply docs/index.rst]: @mistercrunch where do you think this logic should reside? In `index.rst` or `README.md`?

Comment (mistercrunch) [reply docs/index.rst]: both? or one could link to the other?

**Comment by villebro** [comment]
Regarding full or partial names, right now the entries sometimes include the vendor (Amazon Redshift) and sometimes not (BigQuery). Pinot should probably also be listed as "Apache Pinot" with a reference to the official homepage https://pinot.incubator.apache.org/ , like the other Apache projects in the list. Personally, I think it might be better to always include the full name when applicable, e.g. "Google BigQuery". ClickHouse might be an edge case, as I don't think I've ever seen it referred to as "Yandex ClickHouse".

**Comment by john-bodley** [comment]
I’ve tried to use the official name where possible (I missed that Pinot was an Apache project). I’ll go through the list again.

In terms of location I plan to keep the list in the index.rst as there’s merit in having this in the docs and I’ll update the README.md to reference the same list.

**Comment by john-bodley** [comment]
@villebro I find Wikipedia is normally a good source for name definitions. For example Apache Druid is called [Apache Druid](https://en.wikipedia.org/wiki/Apache_Druid) where's Google's BigQuery is simply called [BigQuery](https://en.wikipedia.org/wiki/BigQuery).

**Comment by john-bodley** [comment]
@mistercrunch @villebro I've updated the documentation based on your suggestions.

**Review by villebro — APPROVED** [review]
LGTM. I was also thinking we could add instructions later for how to add a new engine which would specify what documentation changes are to be done in addition to adding a db_engine_specs module.

**Review by mistercrunch — APPROVED** [review]

**Pull request merged**